rb/lib/selenium/webdriver/remote/bridge.rb

```diff
@@ -85,7 +88,12 @@ def initialize(opts = {})
           http_client.server_url = uri
 
           @http = http_client
-          @capabilities = create_session(desired_capabilities)
+          if existing_session
```

Comment:
How about a little bit of simplification?
```patch
-       @capabilities = create_session(desired_capabilities)
+       @capabilites = 
+          if existing_session
+            # Attach to session id rather than creating new session
+            attach_to_session(existing_session)
+          else
+            create_session(desired_capabilities)
+          end
```